tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -53,6 +58,50 @@ public void Cursor_should_not_throw_exception_after_double_close([Values(false,
             }
         }
 
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void KillCursor_should_actually_work(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().Supports(Feature.KillCursorsCommand);
+            var eventCapturer = new EventCapturer().Capture<CommandSucceededEvent>(x => x.CommandName.Equals("killCursors"));
+            using (var client = DriverTestConfiguration.CreateDisposableClient(eventCapturer))
+            {
+                IAsyncCursor<BsonDocument> cursor;
+                var database = client.GetDatabase("test");
+                var collection = database.GetCollection<BsonDocument>(GetType().Name);
+                var documents = new List<BsonDocument>();
+                for (int i = 0; i < 1000; i++)
+                {
+                    documents.Add(new BsonDocument("x", i));
+                }
+
+                if (async)
```

Comment:
My understanding is that the code under test here is confined to the Dispose method of the cursor.  If that is the case, is there any value in including both sync and async paths in what is essentially the setup to the test?  If not, then I'd remove the async path here.  If there is some increased coverage in creating the cursor and/or moving the cursor asynchronously, leave that, but pull out the InsertMany/InsertManyAsync from the conditionals and do it synchronously only

Apologies for not noticing this the first time around.

Reply:
I think you're right and the async case isn't needed. It doesn't appear that creating the cursor both synchronously and asynchronously adds any significant coverage, so I'll remove the async path altogether.